_source/_posts/2022-01-13-spring-cloud-contract.md

```diff
@@ -0,0 +1,893 @@
+
+**Prerequisites:**
+
+Before you get started, you'll need to make sure you have a few tools installed.
+
+- **Okta CLI**: The Okta CLI is an easy way to create projects that use Okta for security. Follow the installation instructions on the [Okta CLI project website](https://cli.okta.com/manual/#installation). You should log in to your existing account or register for a new account using the CLI before continuing with the tutorial.
+
+- **Java 11**: This project uses Java 11. OpenJDK 11 will work just as well.  Instructions are found on the [OpenJDK website](https://openjdk.java.net/install/). OpenJDK can also be installed using [Homebrew](https://brew.sh/). Alternatively, [SDKMAN](https://sdkman.io/) is another excellent option for installing and managing Java versions.
+
+- **HTTPie**: This is a powerful command-line HTTP request utility you'll use to test both Spring Boot resource servers. Install it according to [the docs on their site](https://httpie.org/doc#installation).
+
+## Bootstrap the Producer Project using Spring Initialzr
+
+You'll use the Spring Initializr project to bootstrap the Producer project. Open [this link](https://start.spring.io/#!type=maven-project&language=java&platformVersion=2.6.2&packaging=jar&jvmVersion=11&groupId=com.example&artifactId=producer&name=producer&description=Producer%20project%20for%20Spring%20Boot%20Contract&packageName=com.example.contract&dependencies=web,cloud-contract-verifier,lombok) to access the pre-configured starter project.
+
+{% img blog/spring-cloud-contract/producer-starter-config.png alt:"Producer Starter Configuration" width:"1000" %}{: .center-image }
+
+You are creating a Java project -- Java 11 -- using Maven as the build system. The Spring Boot version was 2.6.2 when I wrote this tutorial. You are including three dependencies: 
+
+- **[Spring Web](https://docs.spring.io/spring-framework/docs/current/reference/html/web.html)**: adds the basic HTTP/web capabilities to build a REST API 
+- **[Contract Verifier](https://spring.io/projects/spring-cloud-contract)**: brings in Spring Cloud Contract project by enabling Consumer Drive Contracts and stub generation
+- **[Lombok](https://projectlombok.org/)**: helper project that will generate some getters, setters, and constructors
+
+When you're ready, click **Generate** and save the zip file in an appropriate parent directory. You'll be making two Spring Boot projects (as mentioned, a producer and a consumer), so it probably makes sense to have a parent directory named something like `Spring Cloud Contract Tutorial` or `spring-cloud-contract-tutorial` (if you dislike spaces in directory names). 
+
+Unzip the file and open the directory in your favorite editor or IDE (I'm an IDE/IntelliJ man myself -- hopefully that's not disappointing to y'all -- because in the immortal words of Buckaroo Banzai "Life is too short for vim.").
```

Comment:
```suggestion
Unzip the file and open the directory in your favorite editor or IDE. (I'm an IDE/IntelliJ man myself -- hopefully that's not disappointing to y'all -- because in the immortal words of Buckaroo Banzai "Life is too short for vim".)
```
One sentence into two. (Once.)